### PR TITLE
Close #12436: refactor JUNIOR_RC_CHAINTYPE enum

### LINKED ISSUE #12436: Refactor JUNIOR_RC_CHAINTYPE to use strong enum

### PR DESCRIPTION
**Pull request by BryanDiLaura**
Let me know if there are any changes that need to be made!

### CODE REVIEW

**Comment by BryanDiLaura** [comment]
Forgot to mention the ticket. #12436

File: src/openrct2/ride/coaster/JuniorRollerCoaster.h [hunk 1]
```diff
@@ -13,11 +13,18 @@
...
 #include "../../paint/tile_element/Paint.TileElement.h"
 
-enum JUNIOR_RC_CHAINTYPE
+// enum JUNIOR_RC_CHAINTYPE
+// {
+//     JUNIOR_RC_CHAIN_NONE,
+//     JUNIOR_RC_CHAIN_FRICTION_WHEELS,
+//     JUNIOR_RC_CHAIN_CHAIN_LIFT,
+// };
+
```
Comment (tupaschoal) [inline src/openrct2/ride/coaster/JuniorRollerCoaster.h:22]:
```suggestion
```
Remove this instead of commenting out.

Comment (BryanDiLaura) [reply src/openrct2/ride/coaster/JuniorRollerCoaster.h:22]: D'oh totally missed that. Fixing now.

File: src/openrct2/ride/coaster/JuniorRollerCoaster.cpp [hunk 2]
```diff
@@ -1622,9 +1622,10 @@ static constexpr const uint32_t junior_rc_track_pieces_diag_60_deg_down_to_25_de
 
 void junior_rc_paint_track_flat(
     paint_session* session, ride_id_t rideIndex, uint8_t trackSequence, uint8_t direction, uint16_t height,
-    const TileElement* tileElement, JUNIOR_RC_CHAINTYPE chainType)
+    const TileElement* tileElement, JuniorRcChainType chainType)
 {
-    uint32_t imageId = junior_rc_track_pieces_flat[chainType][direction] | session->TrackColours[SCHEME_TRACK];
+    uint32_t imageId = junior_rc_track_pieces_flat[static_cast<uint8_t>(chainType)][direction]
+        | session->TrackColours[SCHEME_TRACK];
```
Comment (tupaschoal) [inline src/openrct2/ride/coaster/JuniorRollerCoaster.cpp:1628]:
```suggestion
    uint32_t imageId = junior_rc_track_pieces_flat[EnumValue(chainType)][direction]
        | session->TrackColours[SCHEME_TRACK];
```

**Review by tupaschoal — COMMENTED** [review]
Can you use `EnumValue` in all places that are doing a static_cast currently? It will simplify the code and be more consistent with the latest changes. I've commented one such example. 

(You might have some formatting broken after this, so make sure to run the format check or check the lint result)

File: src/openrct2/ride/coaster/JuniorRollerCoaster.cpp (hunk 2 above)
Comment (BryanDiLaura) [reply src/openrct2/ride/coaster/JuniorRollerCoaster.cpp:1628]: Yeah I can totally do that - definitely agree that it's better. I had no idea that existed, but that's neat!

**Review by tupaschoal — APPROVED** [review]

**Pull request merged**